rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+
+### Considerations when choosing Migration strategy
+- While migration is happening, we want to allow readers to continue reading data. This means, we cannot employ a stop-the-world strategy when we are migrating. 
+All the actions that we are performing as part of table upgrade should not have any side-effects of breaking snapshot isolation for readers.
+- Also, users should have migrated to 0.16.x before upgrading to 1.x. We do not want to add read support for very old versions of hudi in 1.x(for eg 0.7.0). 
+- So, in an effort to bring everyone to latest hudi versions, 1.x reader will have full read capabilities for 0.16.x, but for older hudi versions, 1.x reader may not have full reader support. 
+The reocmmended guideline is to upgrade all readers and writers to 0.16.x. and then slowly start upgrading to 1.x(readers followed by writers). 
+
+Before we dive in further, lets understand the format changes:
+
+## Format changes
+### Table properties
+- Payload class ➝ payload type.
+- hoodie.record.merge.mode is introduced in 1.x. 
+- New metadata partitions could be added (optionally enabled)
+
+### MDT changes
+- New MDT partitions are available in 1.x. MDT schema upgraded.
+- RLI schema is upgraded to hold row positions.
+
+### Timeline:
+- [storage changes] Completed write commits have completed times in the file name(timeline commit files).
+- [storage changes] Completed and inflight write commits are in avro format which were json in 0.x.
+- We are switching the action type for pending clustering from “replace commit” to “cluster”.
+- [storage changes] Timeline ➝ LST timeline. There is no archived timeline in 1.x.
+- [In-memory changes] HoodieInstant changes due to presence of completion time for completed HoodieInstants.
+
+### Filegroup/FileSlice changes:
+- Log file names contain delta commit time instead of base instant time.
+- Log appends are disabled in 1.x. In other words, each log block is already appended to a new log file.
+- File Slice determination logic for log files changed. In 0.x, we have base instant time in log files and its straight forward. 
+In 1.x, we find completion time for a log file and find the base instant time (parsed from base files) for a given HoodieFileGroup, 
+- which has the highest value lesser than the completion time of the log file) of interest. 
+- Log file ordering within a file slice. (in 0.x, we use base instant time ➝ log file versions ➝ write token) to order diff log files. in 1.x, we will be using completion time to order).
+- Rollbacks in 0.x appends a new rollback block (new log file). While in 1.x, rollback will remove the partially failed log files. 
+
+### Log format changes:
+- We have added a new header type, IS_PARTIAL in 1.x.
+
+## Changes to be ported over to 0.16.x to support reading 1.x tables
+
+### What will be supported
+- For features introduced in 0.x, and tables written in 1.x, 0.16.0 reader should be able to provide consistent reads w/o any breakage. 
```

Comment:
So the under-mentioned 1.x features are excluded right?

Reply:
lets expand this more?  for e,g cover types of tables, queries and writer/reader paths to be explicit.. This helps users read this RFC and figure out their plans.